amr-wind/fvm/nonLinearSum.H

```diff
@@ -0,0 +1,166 @@
+#ifndef NONLINEARSUM_H
```

Comment:
@marchdf @mbkuhn thoughts on this?

Reply:
I'm ok with something issue-specific arising here. Especially because using the fvm space is a good way to incorporate modifications to an operator near boundaries. I also think the name itself can be short to reduce verbosity as long as there is some description in the code via comments.

Reply:
I agree in general, but the only boundary/stencil related operator I see in here is the gradient, which might be better computed using `fvm::gradient` instead. The rest of the code isn't `fvm` specific, and looks more like computations using the gradient.